bodhi/tests/server/test_models.py

```diff
@@ -269,6 +269,42 @@ def test_builds(self):
         eq_(self.obj.builds[0].release.name, u'F11')
         eq_(self.obj.builds[0].package.name, u'TurboGears')
 
+    def test_mandatory_days_in_testing_critpath(self):
+        """
+        The Update.mandatory_days_in_testing method should be the configured value
+        for critpath if it is a critpath update.
+        """
+        update = self.obj
+        update.critpath = True
+
+        # Configured value.
+        expected = int(config.get('critpath.stable_after_days_without_negative_karma'))
+
+        eq_(update.mandatory_days_in_testing, expected)
+
+    def test_mandatory_days_in_testing(self):
+        """
+        The Update.mandatory_days_in_testing method should be a positive integer if the
+        mandatory_days_in_testing attribute of release is not truthy.
+        """
+        update = self.obj
+
+        eq_(update.mandatory_days_in_testing, 7)
+
+    def test_days_to_stable_critpath(self):
+        """
+        The Update.days_to_stable() method should return a positive integer depending
+        on the configuration.
+        """
+        update = self.get_update()
+        update.critpath = True
+        update.date_testing = datetime.utcnow() + timedelta(days=-4)
+
+        criptath_days_to_stable = int(
+            config.get('critpath.stable_after_days_without_negative_karma'))
+
+        eq_(update.days_to_stable, criptath_days_to_stable - 4)
+
```

Comment:
These tests cover conditions 1 and 2 that I listed above, but we could still use a third test for the condition where ```release.mandatory_days_in_testing``` is set to 0. You should be able to use ```mock``` to achieve this as I mentioned in my earlier comment.